hbase/src/main/java/org/apache/zeppelin/hbase/HbaseInterpreter.java

```diff
@@ -0,0 +1,152 @@
+import java.util.Properties;
+
+/**
+ * Support for Hbase Shell. All the commands documented here
+ * https://wiki.apache.org/hadoop/Hbase/Shell is supported.
+ *
+ * Requirements:
+ * Hbase Shell should be installed on the same machine. To be more specific, the following dir.
+ * should be available: https://github.com/apache/hbase/tree/master/hbase-shell/src/main/ruby
+ * Hbase Shell should be able to connect to the Hbase cluster from terminal. This makes sure
+ * that the client is configured properly.
+ *
+ * The interpreter takes 3 config parameters:
+ * hbase.home: Root dir. where hbase is installed. Default is /usr/lib/hbase/
+ * hbase.ruby.sources: Dir where shell ruby code is installed.
+ *                          Path is relative to hbase.home. Default: lib/ruby
+ * hbase.irb.load: (Testing only) Default is true.
+ *                      Whether to load irb in the interpreter.
+ */
+public class HbaseInterpreter extends Interpreter {
+  private Logger logger = LoggerFactory.getLogger(HbaseInterpreter.class);
+  private ScriptingContainer scriptingContainer;
+
+  private StringWriter writer;
+
+  static {
+    Interpreter.register("hbase", "hbase", HbaseInterpreter.class.getName(),
+        new InterpreterPropertyBuilder()
+            .add("hbase.home", "/usr/lib/hbase/", "Installation dir. of Hbase")
+            .add("hbase.ruby.sources", "lib/ruby",
+                "Path to Ruby scripts relative to 'hbase.home'")
+            .add("hbase.irb.load", "true", "Load hirb. hirb loads hbase shell. " +
+                "For unit tests it has to be turned off.")
+            .build());
+  }
+
+  public HbaseInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    String hbase_home = getProperty("hbase.home");
+    String ruby_src = getProperty("hbase.ruby.sources");
+    String abs_ruby_src = hbase_home + ruby_src;
+
+    File f = new File(abs_ruby_src);
+    if (!f.exists() || !f.isDirectory()) {
+      throw new InterpreterException("hbase ruby sources is not correctly defined.");
+    }
+
+    logger.info("Home:" + hbase_home);
+    logger.info("Ruby Src:" + ruby_src);
+
+    Properties props = System.getProperties();
+    props.setProperty("hbase.ruby.sources", abs_ruby_src);
+    this.scriptingContainer  = new ScriptingContainer(LocalContextScope.SINGLETON);
+    List<String> paths = new ArrayList<>(Arrays.asList(abs_ruby_src));
+    this.writer = new StringWriter();
+    scriptingContainer.setOutput(this.writer);
+    this.scriptingContainer.setLoadPaths(paths);
+    scriptingContainer.setCompatVersion(org.jruby.CompatVersion.RUBY1_9);
+    if (Boolean.parseBoolean(getProperty("hbase.irb.load"))) {
+      try {
+        InputStream in = getClass().getResourceAsStream("/hbase/bin/hirb.rb");
+        scriptingContainer.runScriptlet(in, "/hbase/bin/hirb.rb");
+        in.close();
+      } catch (NullPointerException | IOException e) {
+        logger.error("Open failed:", e);
```

Comment:
this should rethrow the exception too? I assume the interpreter will not work if this fails